packages/teraslice/test/lib/cluster/services/cluster/backends/kubernetes/k8s-spec.js

```diff
@@ -104,6 +104,33 @@ describe('k8s', () => {
         });
     });
 
+    describe('->nonEmptyList', () => {
+        it('can get list with one item', async () => {
+            nock(_url)
+                .get('/apis/batch/v1/namespaces/default/jobs/')
+                .query({ labelSelector: 'app=teraslice' })
+                .reply(200, { items: ['one'] });
+
+            const jobs = await k8s.nonEmptyList('app=teraslice', 'jobs');
+            expect(jobs.items[0]).toEqual('one');
+        });
+
+        it('throws with an empty list', async () => {
+            nock(_url)
+                .get('/apis/batch/v1/namespaces/default/jobs/')
+                .query({ labelSelector: 'app=teraslice' })
+                .reply(200, { items: [] });
+
+            try {
+                await k8s.nonEmptyList('app=teraslice', 'jobs');
+            } catch (error) {
+                expect(error).toEqual(
+                    Error('Teraslice jobs matching the following selector was not found: app=teraslice (retriable)')
+                );
```

Comment:
If this resolves instead of rejects, the test will pass, jest has an API for handling this more gracefully too.

```suggestion
            await expect(k8s.nonEmptyList('app=teraslice', 'jobs')).rejects.toThrowError('Teraslice jobs matching the following selector was not found: app=teraslice (retriable)');
```

Reply:
Oh, good point, it hadn't occurred to me.

I tried something like that to begin with but couldn't get it working right.

I will try your suggestion and see if it works.

Reply:
Ok, I think I have it now.